spice_agent/utils/memory_size.py

```diff
@@ -0,0 +1,63 @@
+class MemorySize:
+    CONVERSION_FACTORS = {
+        "b": 1 / 8,
+        "B": 1,
+        "KB": 1024,
+        "MB": 1024**2,
+        "GB": 1024**3,
+        "TB": 1024**4,
+        "PB": 1024**5,
+        "KiB": 2**10,
+        "MiB": 2**20,
+        "GiB": 2**30,
+        "TiB": 2**40,
+        "PiB": 2**50,
+    }
+
+    def __init__(self, value, unit):
+        self.value = value
+        self.unit = unit
```

Comment:
let's throw some typing on here.
if it turns out value is always an Int, you can add the `from_string` method in the Init call checking for value as a string to do conversion. this would change your uses of MemorySize.from_string() to ms = MemorySize("some string")